### DISCUSSION TITLE
filterBy with nested object

### DISCUSSION THREAD

**Discussion by azern32** [vincjo/datatables#53] [Q&A] [answered]
is it possible to filter a column by a key in an object?

![Screenshot_2023-08-10_17-46-13](https://github.com/vincjo/datatables/assets/10142910/28aef16f-926f-45a2-a388-8dd9631ae788)

what alternative should I do to achieve the filtering

btw, the data structure for the table was like this
`{
  mID: string,
  customerName: string,
  leadDetail: {
    brand: string,
    phone: string,
    ...,
  }
  ...,
}`

**Comment by vincjo** [comment, marked as answer]
Yes it is possible, by using a callback.

Instead of a key name, `filterBy` can take a function as argument:

```svelte
<ThFilter {handler} filterBy={(row) => row.leadDetail.brand}/>
```

With `DataHandler` instance:
```ts
handler.filter(value, (row) => row.leadDetail.brand)
```

Note that filter by an object works too: `filterBy="leadDetail"` will filter by `brand` + `phone` + ...

**Comment by azern32** [comment]
Ah.... so you can just use a function in that
Thank you so much
